Data/ByteString/Lazy.hs

```diff
@@ -1597,11 +1597,21 @@ illegalBufferSize handle fn sz =
       msg = fn ++ ": illegal ByteString size " ++ showsPrec 9 sz []
 
 -- | Read entire handle contents /lazily/ into a 'ByteString'. Chunks
--- are read on demand, using the default chunk size.
+-- are read on demand, using the default chunk size ('defaultChunkSize').
 --
 -- File handles are closed on EOF if all the file is read, or through
 -- garbage collection otherwise.
 --
+-- Do not use this function inside a 'Control.Exception.bracket' \/
+-- 'System.IO.withFile' \/ 'System.IO.withBinaryFile':
+-- lazy I/O will escape it causing 'Control.Exception.bracket' to close a handle prematurely:
+--
+-- >>> withBinaryFile "foo.txt" ReadMode BL.hGetContents >>= print
+-- "*** Exception: foo.txt: hGetBufSome: illegal operation (handle is closed)
+--
+-- Consider using 'readFile' instead; if you must use 'hGetContents' then open a handle,
```

Comment:
I'm not sure I understand the suggestion to use `readFile` instead. Is the point to avoid the _appearance_ of leaking a file handle? (...and thereby make less obvious that very careful control over evaluation is necessary to avoid _actually_ leaking a file handle?)